### ISSUE TITLE
Sign in to confirm you’re not a bot

### ISSUE THREAD

**Issue by DanielnetoDotCom** [WWBN/AVideo#9242]
#### Problem Description
Since last night and continuing today, users, including admin, are encountering an error when attempting to share, upload, or encode YouTube videos. The error message displayed is:

```
Sorry!
ERROR: [youtube] 6R17SNKEbpY: Sign in to confirm you’re not a bot. This helps protect our community. Learn more.
```

However, the "Learn more" link does not have a hyperlink, and searching online has not provided a solution. 

#### Cause
This issue is a global problem with `yt-dlp`, which is being blocked by YouTube.

#### Current Status
I am actively following the issue for a solution. You can track the progress and updates on this issue at the following link: [GitHub Issue #10128](https://github.com/yt-dlp/yt-dlp/issues/10128).

#### Temporary Solution
As of now, there is no confirmed solution to this problem. Please be patient while the community works on resolving this issue.

Thank you for your understanding and cooperation.

**Comment by eitanwa** [comment]
hey Daniel
Regarding this issue
Not sure we understood
We have no problem uploading or importing files from YouTube
Our systems are up to date
Could you please help us understand what we missed

**Comment by AViSource** [comment]
> hey Daniel Regarding this issue Not sure we understood We have no problem uploading or importing files from YouTube Our systems are up to date Could you please help us understand what we missed

That's weird. This is what's happening when I try to upload a video by importing it. (See screenshot.)
![Screenshot from 2024-07-21 11-06-29](https://github.com/user-attachments/assets/bebdc0e7-8fe1-4110-b40c-050e54789412)

**Comment by DanielnetoDotCom** [comment]
> hey Daniel Regarding this issue Not sure we understood We have no problem uploading or importing files from YouTube Our systems are up to date Could you please help us understand what we missed

Good for you, in those days YouTube is flagging whoever is using yt-dlp as a bot.
also banning users if you try to use cookies on it

**Comment by JoshWho** [comment]
I think it goes of IP they are starting to figure out what servers are being used to pull their vids and banning the ip

**Comment by gbrn** [comment]
@DanielnetoDotCom This problem has now been resolved, I can use **yt-dlp** normally.
I recommend updating the encoder servers. Thanks.

**Comment by para2022** [comment]
same problem for me

**Comment by para2022** [comment]
> @DanielnetoDotCom This problem has now been resolved, I can use **yt-dlp** normally. I recommend updating the encoder servers. Thanks.

still does not work after updating avideo and the encoder. The problem is exactly the same.

**Comment by gbrn** [comment]
@para2022 The encoder does not only work with files available on Github AVideo Encoder.
yt-dlp is responsible for accessing external downloads. So you have to update it on your server.
```
Latest version: stable@2024.08.06 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2024.08.06 from yt-dlp/yt-dlp)
```

`sudo pip install -U yt-dlp`
or
`sudo yt-dlp -U`
or
`sudo curl -L https://github.com/yt-dlp/yt-dlp/releases/latest/download/yt-dlp -o /usr/local/bin/yt-dlp && sudo chmod a+rx /usr/local/bin/yt-dlp`

**Comment by para2022** [comment]
> @para2022 The encoder does not only work with files available on Github AVideo Encoder. yt-dlp is responsible for accessing external downloads. So you have to update it on your server.
> 
> ```
> Latest version: stable@2024.08.06 from yt-dlp/yt-dlp
> yt-dlp is up to date (stable@2024.08.06 from yt-dlp/yt-dlp)
> ```
> 
> `sudo pip install -U yt-dlp` or `sudo yt-dlp -U` or `sudo curl -L https://github.com/yt-dlp/yt-dlp/releases/latest/download/yt-dlp -o /usr/local/bin/yt-dlp && sudo chmod a+rx /usr/local/bin/yt-dlp`

I have updated yt-dlp already to latest and it still doesnt work. Same error message

**Comment by gbrn** [comment]
@para2022 Do a test on your yt-dlp. Here for me it worked normally.
Try checking if your version is stable@2024.08.06, if not, remove and reinstall:
`sudo yt-dlp -v`
```
sudo yt-dlp -v
[debug] Command-line config: ['-v']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.08.06 from yt-dlp/yt-dlp [4d9231208] (zip)
[debug] Python 3.10.12 (CPython aarch64 64bit) - Linux-6.5.0-1027-oracle-aarch64-with-glibc2.35 (OpenSSL 3.0.2 15 Mar 2022, glibc 2.35)
[debug] exe versions: ffmpeg 4.4.2 (setts), ffprobe 4.4.2, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.11.0, brotli-1.0.9, certifi-2020.06.20, mutagen-1.45.1, pyxattr-0.7.2, requests-2.32.3, secretstorage-3.3.1, sqlite3-3.37.2, urllib3-2.2.2, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1830 extractors
```

Test this:

`sudo yt-dlp -f 'bestvideo[height=1080]+bestaudio/best[height=1080]' https://www.youtube.com/watch?v=ucZl6vQ_8Uo`

![image](https://github.com/user-attachments/assets/a3b56423-5605-409b-ada0-ba96c55cfe99)

**Comment by para2022** [comment]
sudo yt-dlp -v
[debug] Command-line config: ['-v']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.08.01 from yt-dlp/yt-dlp [ffd7781d6] (pip)
[debug] Python 3.10.12 (CPython x86_64 64bit) - Linux-5.15.0-117-generic-x86_64-with-glibc2.35 (OpenSSL 3.0.2 15 Mar 2022, glibc 2.35)
[debug] exe versions: ffmpeg 4.4.2 (setts), ffprobe 4.4.2
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2024.07.04, mutagen-1.47.0, requests-2.32.3, secretstorage-3.3.1, sqlite3-3.37.2, urllib3-2.2.2, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Extractor Plugins: oauth2 (YoutubeBaseInfoExtractor), oauth2 (YoutubeClipIE), oauth2 (YoutubeConsentRedirectIE), oauth2 (YoutubeFavouritesIE), oauth2 (YoutubeIE), oauth2 (YoutubeMusicSearchURLIE), oauth2 (YoutubeNotificationsIE), oauth2 (YoutubeSearchDateIE), oauth2 (YoutubeSearchIE), oauth2 (YoutubeSearchURLIE), oauth2 (YoutubeTabBaseInfoExtractor), oauth2 (YoutubeTabIE)
[debug] Plugin directories: ['/usr/local/lib/python3.10/dist-packages/yt_dlp_plugins']
[debug] Loaded 1830 extractors

Usage: yt-dlp [OPTIONS] URL [URL...]

yt-dlp: error: You must provide at least one URL.
Type yt-dlp --help to see a list of all options.

**Comment by AViSource** [comment]
> sudo yt-dlp -v [debug] Command-line config: ['-v'] [debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8 [debug] yt-dlp version stable@2024.08.01 from yt-dlp/yt-dlp [ffd7781d6] (pip) [debug] Python 3.10.12 (CPython x86_64 64bit) - Linux-5.15.0-117-generic-x86_64-with-glibc2.35 (OpenSSL 3.0.2 15 Mar 2022, glibc 2.35) [debug] exe versions: ffmpeg 4.4.2 (setts), ffprobe 4.4.2 [debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2024.07.04, mutagen-1.47.0, requests-2.32.3, secretstorage-3.3.1, sqlite3-3.37.2, urllib3-2.2.2, websockets-12.0 [debug] Proxy map: {} [debug] Request Handlers: urllib, requests, websockets [debug] Extractor Plugins: oauth2 (YoutubeBaseInfoExtractor), oauth2 (YoutubeClipIE), oauth2 (YoutubeConsentRedirectIE), oauth2 (YoutubeFavouritesIE), oauth2 (YoutubeIE), oauth2 (YoutubeMusicSearchURLIE), oauth2 (YoutubeNotificationsIE), oauth2 (YoutubeSearchDateIE), oauth2 (YoutubeSearchIE), oauth2 (YoutubeSearchURLIE), oauth2 (YoutubeTabBaseInfoExtractor), oauth2 (YoutubeTabIE) [debug] Plugin directories: ['/usr/local/lib/python3.10/dist-packages/yt_dlp_plugins'] [debug] Loaded 1830 extractors
> 
> Usage: yt-dlp [OPTIONS] URL [URL...]
> 
> yt-dlp: error: You must provide at least one URL. Type yt-dlp --help to see a list of all options.

We have also applied all of the above mentioned steps and gotten the same results as you are getting.

**Comment by gbrn** [comment]
For me it stopped working again, the last update is not working is:
Latest version: stable@2024.08.06 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2024.08.06 from yt-dlp/yt-dlp)

**Comment by DanielnetoDotCom** [comment]
Hi

I just sent some update to youtube can you guys update your Encoder and signing on youtube before download?

![image](https://github.com/user-attachments/assets/9188b1db-a74b-411e-9fc2-9743797dfc94)

**Comment by AViSource** [comment]
I have updated my Encoder and completed the authentication step. But before we even try to upload or share content. I decided to take a look at my logfile.  And sure enough, there's an error. I want to download it and share it. How do I download my log file?

**Comment by para2022** [comment]
It doesnt work for me 

> Hi
> 
> I just sent some update to youtube can you guys update your Encoder and signing on youtube before download?
> 
> ![image](https://private-user-images.githubusercontent.com/18009293/357177022-9188b1db-a74b-411e-9fc2-9743797dfc94.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM2MDcxMjEsIm5iZiI6MTcyMzYwNjgyMSwicGF0aCI6Ii8xODAwOTI5My8zNTcxNzcwMjItOTE4OGIxZGItYTc0Yi00MTFlLTlmYzItOTc0Mzc5N2RmYzk0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE0VDAzNDAyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA3NjI0NDZkZTlhY2JkNTkxNGQzNzY0ZWI3NmEwZTllMTRhNzhjYzkxNTllMTg3ZjY2MmQ0Nzc5MGM2ZDhmZDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.kBQkM9eFblXQpSH2IM3UjKpdl-pD1y0fV1WIP3Rg6ig)

This doesnt work me. Even with the fix i still get the same error:

ERROR: [youtube+oauth2] tMkUIHnv6Yw: Sign in to confirm you’re not a bot. This helps protect our community. Learn more

**Comment by para2022** [comment]
[14-Aug-2024 04:39:50 Europe/London] [2024-08-14 04:39:50] getTitleFromLink: Get Title Error: LC_ALL=en_US.UTF-8 yt-dlp   --no-check-certificate --no-playlist --force-ipv4 --skip-download -e 'https://www.youtube.com/watch?v=tMkUIHnv6Yw'
Array
(
    [0] => ERROR: [youtube+oauth2] tMkUIHnv6Yw: Sign in to confirm you’re not a bot. This helps protect our community. Learn more
)

[14-Aug-2024 04:39:50 Europe/London] youtubeDl::addVideo We could not get the title () of your video (https://www.youtube.com/watch?v=tMkUIHnv6Yw)
[14-Aug-2024 04:39:50 Europe/London] PHP Warning:  Undefined variable $encoders_ids in /home/encoder1/public_html/encoder1/objects/functions.php on line 1508
[14-Aug-2024 04:39:50 Europe/London] PHP Stack trace:
[14-Aug-2024 04:39:50 Europe/London] PHP   1. {main}() /home/encoder1/public_html/encoder1/view/youtubeDl.json.php:0
[14-Aug-2024 04:39:50 Europe/London] PHP   2. addVideo($link = 'https://www.youtube.com/watch?v=tMkUIHnv6Yw', $streamers_id = 1, $title = NULL) /home/encoder1/public_html/encoder1/view/youtubeDl.json.php:53
[14-Aug-2024 04:39:50 Europe/London] PHP Warning:  Trying to access array offset on null in /home/encoder1/public_html/encoder1/objects/functions.php on line 1508
[14-Aug-2024 04:39:50 Europe/London] PHP Stack trace:
[14-Aug-2024 04:39:50 Europe/London] PHP   1. {main}() /home/encoder1/public_html/encoder1/view/youtubeDl.json.php:0
[14-Aug-2024 04:39:50 Europe/London] PHP   2. addVideo($link = 'https://www.youtube.com/watch?v=tMkUIHnv6Yw', $streamers_id = 1, $title = NULL) /home/encoder1/public_html/encoder1/view/youtubeDl.json.php:53

**Comment by DanielnetoDotCom** [comment]
make sure you sign in properly, I am not seeing the accessToken in your log. 

try also to share more from your logs

**Comment by para2022** [comment]
It shows me as properly logged in on my Google account and shows as signed in on the encoder

**Comment by DanielnetoDotCom** [comment]
Ok, so share more from your logs, 

if you have signed in in your yt-dlp should have something like `--add-header "Authorization: Bearer...`

**Comment by gbrn** [comment]
I would like to report my tests... doing Google authentication, it worked perfectly.
I hope that soon this will no longer be necessary.

**Comment by DanielnetoDotCom** [comment]
> It shows me as properly logged in on my Google account and shows as signed in on the encoder

@para2022 please make a git pull again on the encoder

**Comment by para2022** [comment]
That last update fixed it. Thanks a lot dan :)

**Comment by AViSource** [comment]
> > It shows me as properly logged in on my Google account and shows as signed in on the encoder
> 
> @para2022 please make a git pull again on the encoder

I am still experiencing the same problem.

Encoder has been updated, I am logged in to my Google account and shows me as signed in on the encoder as well. yet I get the following error when I attempt to share:

Sorry!
WARNING: [youtube] YouTube said: ERROR - Request had invalid authentication credentials. Expected OAuth 2 access token, login cookie or other valid authentication credential. See https://developers.google.com/identity/sign-in/web/devconsole-project., WARNING: [youtube] HTTP Error 401: Unauthorized. Retrying (1/3)..., WARNING: [youtube] YouTube said: ERROR - Request had invalid authentication credentials. Expected OAuth 2 access token, login cookie or other valid authentication credential. See https://developers.google.com/identity/sign-in/web/devconsole-project., WARNING: [youtube] HTTP Error 401: Unauthorized. Retrying (2/3)..., WARNING: [youtube] YouTube said: ERROR - Request had invalid authentication credentials. Expected OAuth 2 access token, login cookie or other valid authentication credential. See https://developers.google.com/identity/sign-in/web/devconsole-project., WARNING: [youtube] HTTP Error 401: Unauthorized. Retrying (3/3)..., WARNING: [youtube] YouTube said: ERROR - Request had invalid authentication credentials. Expected OAuth 2 access token, login cookie or other valid authentication credential. See https://developers.google.com/identity/sign-in/web/devconsole-project., WARNING: [youtube] Unable to download API page: HTTP Error 401: Unauthorized (caused by ), WARNING: [youtube] YouTube said: ERROR - Request had invalid authentication credentials. Expected OAuth 2 access token, login cookie or other valid authentication credential. See https://developers.google.com/identity/sign-in/web/devconsole-project., WARNING: [youtube] HTTP Error 401: Unauthorized. Retrying (1/3)..., WARNING: [youtube] YouTube said: ERROR - Request had invalid authentication credentials. Expected OAuth 2 access token, login cookie or other valid authentication credential. See https://developers.google.com/identity/sign-in/web/devconsole-project., WARNING: [youtube] HTTP Error 401: Unauthorized. Retrying (2/3)..., WARNING: [youtube] YouTube said: ERROR - Request had invalid authentication credentials. Expected OAuth 2 access token, login cookie or other valid authentication credential. See https://developers.google.com/identity/sign-in/web/devconsole-project., WARNING: [youtube] HTTP Error 401: Unauthorized. Retrying (3/3)..., WARNING: [youtube] YouTube said: ERROR - Request had invalid authentication credentials. Expected OAuth 2 access token, login cookie or other valid authentication credential. See https://developers.google.com/identity/sign-in/web/devconsole-project., WARNING: [youtube] Unable to download API page: HTTP Error 401: Unauthorized (caused by ), WARNING: Only images are available for download. use --list-formats to see them, ERROR: [youtube] UFDQGBfCd1M: Requested format is not available. Use --list-formats for a list of available formats

What step did I miss? 
![2024-08-28 23_12_55-Greenshot](https://github.com/user-attachments/assets/ff1a4975-153f-40dc-8563-aad658670b85)
![2024-08-28 23_12_49-Greenshot](https://github.com/user-attachments/assets/4c642edd-ad12-442c-ae49-080b6b0b863c)
![2024-08-28 23_12_41-Greenshot](https://github.com/user-attachments/assets/d6131c6c-ef58-4c12-ad18-798b2c5c159d)
![2024-08-28 23_12_22-Encoder - Vivaldi](https://github.com/user-attachments/assets/19a2bcc9-29e9-454a-8cae-6ee01e939a3b)

**Comment by DanielnetoDotCom** [comment]
@AViSource  What domain is yours?

what youtube URL are you trying to download?

**Comment by AViSource** [comment]
> @AViSource What domain is yours?
> 
> what youtube URL are you trying to download?
> 
> I guess you are from ubaantu, if yes I will look it direcly

Yes, I am from ubaantu

**Comment by DanielnetoDotCom** [comment]
What exactly link are you trying to download?

**Comment by AViSource** [comment]
> What exactly link are you trying to download?
Oh, sorry. This is it.
https://www.youtube.com/watch?v=tKELDE5Hazg

**Comment by DanielnetoDotCom** [comment]
I went to your encoder and signin with my Google user 

![image](https://github.com/user-attachments/assets/fc8cef50-65e6-401c-93a5-b903ec0dec0a)


and it works for me  

![image](https://github.com/user-attachments/assets/102f2856-0d59-4e87-83e5-10830f40b85e)

soon you will see it in your library

![image](https://github.com/user-attachments/assets/8d5ce927-3199-4bf8-b9c3-6cc883ae3985)

**Comment by AViSource** [comment]
> I went to your encoder and signin with my Google user
> 
> ![image](https://private-user-images.githubusercontent.com/18009293/362463444-fc8cef50-65e6-401c-93a5-b903ec0dec0a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ4ODQ2MjIsIm5iZiI6MTcyNDg4NDMyMiwicGF0aCI6Ii8xODAwOTI5My8zNjI0NjM0NDQtZmM4Y2VmNTAtNjVlNi00MDFjLTkzYTUtYjkwM2VjMGRlYzBhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI4VDIyMzIwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY2YjA2MzczNWM3YTkyMGIzMmUyZWY4ZGY5ZDM3N2YxNDIwYmY5NmRmMjBlNzc4MTgyZTE1YWM1NmE1NzA2NWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.dhAUlTA8ua2cnddf-E_sbYncdfl6ZYbxwU9fdL5H204)
> 
> and it works for me
> 
> ![image](https://private-user-images.githubusercontent.com/18009293/362463536-102f2856-0d59-4e87-83e5-10830f40b85e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ4ODQ2MjIsIm5iZiI6MTcyNDg4NDMyMiwicGF0aCI6Ii8xODAwOTI5My8zNjI0NjM1MzYtMTAyZjI4NTYtMGQ1OS00ZTg3LTgzZTUtMTA4MzBmNDBiODVlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI4VDIyMzIwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc0NmExMjIzMWZhM2RkMzVjMTMwYzI2Yjg2YzZlZmFiZjQ4Y2EyNDJjZTYzYTI4MWY3NGRmMzFiM2U5MzU0NzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.sfmKKqQFVqTULl2tRN6afpLBkgYyTJDYHkjwCyKMlHM)
> 
> soon you will see it in your library
> 
> ![image](https://private-user-images.githubusercontent.com/18009293/362463694-8d5ce927-3199-4bf8-b9c3-6cc883ae3985.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ4ODQ2MjIsIm5iZiI6MTcyNDg4NDMyMiwicGF0aCI6Ii8xODAwOTI5My8zNjI0NjM2OTQtOGQ1Y2U5MjctMzE5OS00YmY4LWI5YzMtNmNjODgzYWUzOTg1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI4VDIyMzIwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJjODNlMDVkMWNiMzVjYTM5ODQ4MGNiZjhkOTdiOGZmMzlhYWQ3YTQyYzQ5MzYzYTgyZmZkNjgzZGIzOTUwZDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.0VOCav5iNfJq36JpskgOenP1c26kXkZttMUUfy9eTKY)

I'm happy that it worked. I  will try to download / share another video. In my attempts, I selected "Import Video", pasted the url link, selected a "Category" and clicked "Share" before getting the error. I give an update.

**Comment by AViSource** [comment]
This is weird. I am still getting the same error. There is something off somewhere.

**Comment by DanielnetoDotCom** [comment]
where you sign in on google, try to delete and add it again. 

from your logs, looks like you did not authorize it properly

**Comment by AViSource** [comment]
> where you sign in on google, try to delete and add it again.
> 
> from your logs, looks like you did not authorize it properly

Ok so I deleted it and added it again. But, it did not work. So, used another one of our accounts and it worked. I am happy that it worked; However, I am concerned about why it worked with one account but not the other.

**Comment by DanielnetoDotCom** [comment]
it is probabaly a google thing.

**Comment by gbrn** [comment]
@DanielnetoDotCom Danny the encoder is giving me a different error now.

`WARNING: [youtube] YouTube said: ERROR - Request contains an invalid argument., WARNING: [youtube] HTTP Error 400: Bad Request. Retrying (1/3)..., WARNING: [youtube] YouTube said: ERROR - Request contains an invalid argument., WARNING: [youtube] HTTP Error 400: Bad Request. Retrying (2/3)..., WARNING: [youtube] YouTube said: ERROR - Request contains an invalid argument., WARNING: [youtube] HTTP Error 400: Bad Request. Retrying (3/3)..., WARNING: [youtube] YouTube said: ERROR - Request contains an invalid argument., WARNING: [youtube] Unable to download API page: HTTP Error 400: Bad Request (caused by ), ERROR: [youtube] 9thU1yw2oU0: Video unavailable. This content isn’t available.`

![image](https://github.com/user-attachments/assets/7c7ca447-0256-4ade-8db6-45d2d6b34cbb)

LOG:
```
    [4] => WARNING: [youtube] YouTube said: ERROR - Request contains an invalid argument.
    [5] => WARNING: [youtube] HTTP Error 400: Bad Request. Retrying (3/3)...
    [6] => WARNING: [youtube] YouTube said: ERROR - Request contains an invalid argument.
    [7] => WARNING: [youtube] Unable to download API page: HTTP Error 400: Bad Request (caused by <HTTPError 400: Bad Request>)
    [8] => ERROR: [youtube] 9thU1yw2oU0: Video unavailable. This content isn’t available.
)

[29-Oct-2024 18:43:28 America/Sao_Paulo] youtubeDl::addVideo We could not get the title () of your video (https://www.youtube.com/watch?v=9thU1yw2oU0)
[29-Oct-2024 18:43:28 America/Sao_Paulo] PHP Warning:  Undefined variable $encoders_ids in /var/www/html/AVideo-Encoder/objects/functions.php on line 1508
[29-Oct-2024 18:43:28 America/Sao_Paulo] PHP Warning:  Trying to access array offset on null in /var/www/html/AVideo-Encoder/objects/functions.php on line 1508

```

**Comment by DanielnetoDotCom** [comment]
The error is really this `Video unavailable. This content isn’t available.` probably a location restriction

**Comment by gbrn** [comment]
@DanielnetoDotCom That's not it, I've already tested other URLs. Probably a new block.
Can you test this URL on your side?
https://www.youtube.com/watch?v=ucZl6vQ_8Uo

**Comment by DanielnetoDotCom** [comment]
You can try our public encoders

https://encoder1.wwbn.net/

**Comment by gbrn** [comment]
> The error is really this `Video unavailable. This content isn’t available.` probably a location restriction

@DanielnetoDotCom Danny, the problem was very simple.... the problem was with my GOOGLE account. I changed the account and it worked.

**Comment by gbrn** [comment]
@DanielnetoDotCom Danny, the problem happened 2 weeks ago. I changed the Google Account and the problem was solved. Now the problem is back, I changed the Google account and the problem did not solve.

![image](https://github.com/user-attachments/assets/841f5cf8-6229-497b-b35c-f559d1612ef1)

**Comment by DanielnetoDotCom** [comment]
I can see the video is unavailable, maybe it is not available on your server country location.

**Comment by gbrn** [comment]
Danny, the problem is not related to the country location, I tested two servers in different location:

Using your servers encoder1.wwbn.net:
I notice that as I change the video, the errors are also different:

`WARNING: Only images are available for download. use --list-formats to see them, ERROR: [youtube] QfQecOHhspc: Requested format is not available. Use --list-formats for a list of available formats`

or

`ERROR: [youtube] ucZl6vQ_8Uo: Video unavailable. This content isn’t available.`

Now using a server in Brazil, in the same region as mine:

`ERROR: [youtube] C0DPdy98e4c: Video unavailable. This content isn’t available.`

**Comment by DanielnetoDotCom** [comment]
But this message is from youtube, it is some restriction on youtube side. 

there is nothing I can do on it

BTW what URL is that?

**Comment by gbrn** [comment]
I think Youtube updated something and yt-dlp hasn't been updated yet...

URLs:
https://www.youtube.com/watch?v=ucZl6vQ_8Uo
https://www.youtube.com/watch?v=C0DPdy98e4c

**Comment by DanielnetoDotCom** [comment]
Acctually all the videos seems to be the same, let me research a bit

**Comment by gbrn** [comment]
Unfortunately we will have to wait for the next yt-dlp update.

**Comment by DanielnetoDotCom** [comment]
### Why Cookies Aren't a Great Solution for AVideo Encoder

Using cookies for YouTube authentication has too many downsides:

1. **Too Much Work for Users**: Users have to manually export cookies from their browser and upload them. This is time-consuming and not user-friendly.

2. **Short Lifespan**: Cookies can expire or get rotated by YouTube often, meaning users will have to repeat the process regularly.

3. **Security Risks**: Cookies contain sensitive account info. Mishandling them or storing them improperly could expose users to risks.

4. **No Automation**: There's no way to automatically refresh or update cookies, so when they expire, the process breaks.

In short, cookies are unreliable, inconvenient, and not secure enough for a smooth experience. A better solution would focus on automation or alternative methods.

**Comment by gbrn** [comment]
Apparently yt-dlp will stop using OAuth... it seems that YT is using an A/B test, which does not allow it to work. The only indication is to use Cookies...

Another solution I see is to manually download the videos and upload them.

**Comment by DanielnetoDotCom** [comment]
How do you manually download them?

**Comment by gbrn** [comment]
I use those YouTube download sites. One that has worked for me is:
https://y2down.cc/
It's more work, but it works for me temporarily.
Several of these sites are also having problems.
There are some software that may work too, such as: 4K Video Downloader.

**Comment by DanielnetoDotCom** [comment]
I will release a solution soon

check this https://github.com/WWBN/AVideo-Encoder/wiki/Switching-to-Python-Script-for-YouTube-Downloads

**Comment by DanielnetoDotCom** [comment]
I am not sure if the signup still required for title and thumbnail, but I will probably switch all to python

**Comment by JoshWho** [comment]
> I will release a solution soon
> 
> check this https://github.com/WWBN/AVideo-Encoder/wiki/Switching-to-Python-Script-for-YouTube-Downloads

Server is going to need Google Chrome installed and a path to the cookies for dlp to work.

**Comment by DanielnetoDotCom** [comment]
The solution I did does not work on servers, only worked on my local PC, probably because of the IP.

I tested in different servers, but on local PC works fine

I am not sure what to do now.

**Comment by ClearStreamDevTeam** [comment]
I don't think this has to do with Daniel or the script. I had the same issue and it appears to be coming from Youtubes side. If you are a creator they have live chat and they are able to assist

**Comment by eitanwa** [comment]
> The solution I did does not work on servers, only worked on my local PC, probably because of the IP.
> 
> I tested in different servers, but on local PC works fine
> 
> I am not sure what to do now.

Hey Daniel
Now that I think about it
Isn't this related to the use of HLS?
It started on YouTube since the use of this technology am I right?
As on our platform now HLS videos cannot be downloaded
We also can't import our HLS files

**Comment by DanielnetoDotCom** [comment]
Hi @eitanwa I do not think so, we are talking about youtube. not avideo. 
not sure if youtube protects it, and also I always try to download the youtube mp4 version.

**Comment by DanielnetoDotCom** [comment]
## Solution Found! YouTube Downloads Working Again

After extensive testing, we found a solution that makes YouTube downloads work again. The key issue is that **YouTube now requires a JavaScript runtime** to solve the "n challenge" protection.

### Requirements

1. **Update yt-dlp to the latest version:**
```bash
sudo yt-dlp -U
# or
sudo pip install -U yt-dlp
# or
sudo curl -L https://github.com/yt-dlp/yt-dlp/releases/latest/download/yt-dlp -o /usr/local/bin/yt-dlp && sudo chmod a+rx /usr/local/bin/yt-dlp
```

2. **Install Deno (JavaScript runtime) - Ubuntu/Debian:**
```bash
# Install Deno
curl -fsSL https://deno.land/install.sh | sh

# Copy to global location (IMPORTANT - so www-data can access it)
sudo cp ~/.deno/bin/deno /usr/local/bin/deno
sudo chmod 755 /usr/local/bin/deno

# Verify www-data can run it
sudo -u www-data /usr/local/bin/deno --version
```

**Alternative:** If you prefer Node.js instead of Deno:
```bash
sudo apt update
sudo apt install -y nodejs
```

### New Encoder Features

I've added:
- **Automatic JS runtime check** before attempting YouTube downloads
- **Clear error messages** when Deno/Node.js is not installed
- **Diagnostic script**: `php install/checkJSRuntime.php` - run this to check your setup

### Why This Happens

YouTube implemented a JavaScript-based protection ("n challenge") that encrypts video URLs. Without a JS runtime (Deno or Node.js), yt-dlp cannot decrypt the URLs and shows "This video is not available".

More info: https://github.com/yt-dlp/yt-dlp/wiki/EJS

---

**Tested and confirmed working on February 10, 2026.**

Similar code found with 1 license type
